caddyhttp/proxy/proxy.go

```diff
@@ -59,7 +61,10 @@ type UpstreamHost struct {
 	WithoutPathPrefix string
 	ReverseProxy      *ReverseProxy
 	Fails             int32
-	Unhealthy         bool
+	// This is an int32 so that we can use atomic operations to do concurrent
+	// reads & writes to this value.  The default value of 0 indicates that it
+	// is healthy and any non-zero value indicates unhealthy.
+	Unhealthy int32
```

Comment:
Oh neat, I hadn't considered that, actually.  I just selected int32 since it seemed large enough to hold the single bit value that we are putting in there. :-)